include/envoy/network/transport_socket.h

```diff
@@ -226,6 +229,11 @@ class TransportSocketFactory {
    */
   virtual TransportSocketPtr
   createTransportSocket(TransportSocketOptionsSharedPtr options) const PURE;
+
+  /**
+   * @return bool whether the transport socket will use proxy protocol options.
+   */
+  virtual bool usesProxyProtocolOptions() const { return false; }
```

Comment:
tap transport socket would need return underlying transport socket value instead of false.